compiler/rustc_abi/src/layout.rs

```diff
@@ -1103,6 +1103,9 @@ fn univariant<
         inverse_memory_index.into_iter().map(|it| it.index() as u32).collect()
     };
     let size = min_size.align_to(align.abi);
+    if size.bytes() >= dl.obj_size_bound() {
```

Comment:
This feels like an arbitrary place. And now we have two of these checks. Can we instead do this in a single place before producing any `LayoutS`? Or maybe even making `align_to` and other size operations fallible?

Reply:
I don't understand the structure of this code, so I'm afraid I don't know where else to put it. This is in `univeriant` which sounds reasonably canonical? The other one is somewhere in the enum code I think. :shrug:

If you can tell me another place to put it I am happy yo move it, but me picking one random place after the other until you are happy does not sound like a good strategy. ;)

> Or maybe even making align_to and other size operations fallible?

Currently Size has two kinds of operations -- some check the obj-size-bound and some don't. I don't know why, but maybe some part of the compiler needs to work with larger sizes? No idea.

I can make `align_to` take a `dl` so that it can do the check. But that makes little sense when Size implements `*` and `+` which can't do the check since they don't have a `dl`.

Reply:
> Currently Size has two kinds of operations -- some check the obj-size-bound and some don't. I don't know why, but maybe some part of the compiler needs to work with larger sizes? No idea.

most of these are likely just ad-hoc



> If you can tell me another place to put it I am happy yo move it, but me picking one random place after the other until you are happy does not sound like a good strategy. ;)

😆 I'm not sure where to put it either, needs experimentation. So... just leave a `FIXME(oli-obk): deduplicate and harden these checks` on both checks 

r=me with fixme comment